src/app/child-dev-project/children/children-bmi-dashboard/children-bmi-dashboard.component.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { RouterTestingModule } from "@angular/router/testing";
+import { HealthCheck } from "app/child-dev-project/health-checkup/model/health-check";
+import { of } from "rxjs";
+import { ChildrenService } from "../children.service";
+import { Child } from "../model/child";
+
+import { ChildrenBmiDashboardComponent } from "./children-bmi-dashboard.component";
+
+describe("ChildrenBmiDashboardComponent", () => {
+  let component: ChildrenBmiDashboardComponent;
+  let fixture: ComponentFixture<ChildrenBmiDashboardComponent>;
+  const mockChildrenService: jasmine.SpyObj<ChildrenService> = jasmine.createSpyObj(
+    "mockChildrenService",
+    ["getHealthChecksOfChild", "getChildren"]
+  );
+
+  beforeEach(() => {
+    mockChildrenService.getChildren.and.returnValue(of([]));
+    TestBed.configureTestingModule({
+      declarations: [ChildrenBmiDashboardComponent],
+      imports: [RouterTestingModule.withRoutes([])],
+      providers: [{ provide: ChildrenService, useValue: mockChildrenService }],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ChildrenBmiDashboardComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it("should create", () => {
+    expect(component).toBeTruthy();
+  });
+
+  it("should load the BMI data for the childs, but only display the unhealthy one", (done) => {
+    const testChild = new Child("testID");
+    const HealthCheck1 = new HealthCheck("hc1");
```

Comment:
It would make the test a little easier to understand if you call the variables according to what they represent, e.g. `childWithLowBmi`, `oldGoodHealthCheck`, `newBadHealthCheck`,... Because then the assertions in the end are easier to understand.